### ISSUE TITLE
Search doesn't show the query if no results found

### ISSUE THREAD

**Issue by teo1978** [phpmyadmin/phpmyadmin#14542]
### Steps to reproduce
1. Log in, select a database and a table
2. go to Search
3. Enter some search criteria that DO yield results and hit Go
4. Repeat, but this time use search criteria that DON'T yield any result

### Expected behaviour
In both cases (3) and (4), the exact same thing should happen: that both the query and the results are shown. In case (4) the results are "MySQL returned an empty result set (i.e. zero rows)" while in case (3) there are actual results.

### Actual behaviour
When there are results, the query is shown too (and you can edit it, etc).
When there are no results, the query is not shown, which doesn't make any sense and is tremendously annoying.

Whatever those of the online demo are: 
https://demo.phpmyadmin.net/

**Comment by ashraysinha6069** [comment]
i would like to work on this issue...
and some guidance on where to start would be great

**Comment by lem9** [comment]
I don't find that annoying. The assumption is that you would like to redo your search, and for this the "Show search criteria" permits to redisplay the search area where you can refine your criteria.

**Comment by teo1978** [comment]
> The assumption is that you would like to redo your search

The assumption is wrong, and you shouldn't be making assumptions.

The reasons for showing the query when the result set is empty are exactly the same as when the result is not empty. Showing the query in one case but not the other doesn't make any sense.

Reasons for being able to see the query and edit it (other than just editing the search criteria):
- I want to know what the query is for this search (maybe I am a beginner and I used the search tool in order to learn what the query is for doing such a search; or maybe I just want to copy and paste it to my code or on StackOverflow)
- I want to make a slightly different query that is not just a different search criteria, i.e. edit the query freely (I was too lazy to write it by hand the first time, but now I want to edit it by hand; or I realze what I need is more complex than what the search form provides; or I used the search just as a starting point...)
- I want to review the query because I expected it to find results so I want to have a quick glance at it to see if there's anything wrong in it (could be easier for me to spot it by looking at the query than by looking at the search criteria in the UI)
- ...

And countless more, but I shouldn't need to mention these, because they are the same use cases as when the result set is not empty, and in that case you do show the query.

**Comment by lem9** [comment]
Pardon me but after 20 years using phpMyAdmin, I'll make assumptions whenever I feel to. My assumption is based on my perceived needs of the majority of users, who, I believe, don't want to type SQL queries or review them.

**Comment by teo1978** [comment]
So you wouldn't show the query when there _are_ search results, either.
Nor when one is just browsing a table's contents. Or after deleting a row.

Nor basically EVERYWHERE, because that's what phpmyadmin currently does, show the performed queries practically everywhere, whatever you do through the interface, except in this specific case, when you do a search _and_ the search result is empty.

> after 20 years using phpMyAdmin, I'll make assumptions whenever I feel to.

Whatever.

The relevant thing here is that showing a query has only advantages over not showing it. Showing it satisfies everyone, both those who are interested in it and those who are not (who can just ignore it). Not showing it only satisfies those who don't need it.

**Comment by lem9** [comment]
You might have a good point, but I wonder why, if I remember correctly, nobody else mentioned this.

**Comment by ltguide** [comment]
I'll just chime in here. I also find it strange that the query is not displayed when there are no results returned. If anything, I would expect the "search criteria" to automatically be visible for quicker editing by non-SQL users.

But as a "power user" I would appreciate having the SQL queries always displayed.

**Comment by williamdes** [comment]
@ibennetch @lem9 Is this issue a feature request or should it be closed ?

**Comment by teo1978** [comment]
@williamdes neither. It's a bug and there's no reason to close it unless it's fixed.

**Comment by benjamw** [comment]
My two cents...

When using GUIs to do certain things, it's often at a price, and that price is not knowing what is going on in the background. So PMA admins have made the decision to hide that from the typical user. BUT... I would much rather see an invalid query so I can quickly figure out why it failed to return results than to see a valid query that gave me what I was looking for.

I'm also of the mind that when working with a GUI that replicates CLI commands, there should be a place to see EVERY command/query given so that it can be replicated, and if something goes wrong, debugged.

Showing the query in one place when screen real estate is already used up by search boxes and result sets instead of another place where that space is free for use seems a bit backward.

**Comment by teo1978** [comment]
> So PMA admins have made the decision to hide that from the typical user

Have they, though? I wonder if this was actually by accident.

**Comment by williamdes** [comment]
> I wonder if this was actually by accident.

If someone has some time to search in git history ?

**Comment by teo1978** [comment]
Not that it's very relevant anyway. If it was a decision, it was clearly a wrong one.

**Comment by rishikesh1410** [comment]
Hello Everyone , I also find it strange .
When you click on the sql tab and enter a query that doesn't yield a result , it display that query with empty result but when you click on the search tab and search a query that doesn't yield a result than it doesn't display that query .
I want to ask that Can I also work on this issue ?

**Comment by williamdes** [comment]
> Hello Everyone , I also find it strange .
> When you click on the sql tab and enter a query that doesn't yield a result , it display that query with empty result but when you click on the search tab and search a query that doesn't yield a result than it doesn't display that query .
> I want to ask that Can I also work on this issue ?

I maybe missed the notification.. 
Yes, anyone can work on this one but I will want to have some research done in the history to understand if it is a big or a design decision.

**Comment by williamdes** [comment]
Hi everyone.
TLDR;
Time has passed and I am fed up to be frustrated not to have the SQL query available after searching and finding nothing.
I prefer to craft a query that will not work and then edit it rather that wasting time to write the full query for the pleasure of writing queries.

I want to have a "show the query" button like we have everywhere else.

:ping_pong: @teo1978

**Comment by Patfreeze** [comment]
This will be a nice to have. 👍 
I work on different servers with the same database structure. Sometimes a server can return result but the other one not.
So when the query not show up, I make the inverse search for the query (lazy i know) and copy it to paste in other servers. 😄 

With this message I will also give you a big thanks for all your work.

**Comment by Izurii** [comment]
I'm getting this issue to work on, and I'm thinking that I made some progress.

This is what the issue was talking about, a query that return zero rows doesn't show the query itself neither options to edit the query etc:

![31dfdd244881965f0a12df52b336062c](https://user-images.githubusercontent.com/46232520/89929912-4a10e580-dbe0-11ea-8d94-ecac8e66ad18.png)

I just did that, is this what supposed to look like?

![0484d7963b338f2833e563e35f0c646b](https://user-images.githubusercontent.com/46232520/89929664-e8e91200-dbdf-11ea-992c-9a630957572d.png)

**Comment by Patfreeze** [comment]
That is what we are looking for! Nice job.

**Comment by Izurii** [comment]
I noticed something, after the first search that gave zero rows as result, the next one that you do by using ``Edit`` or ``Edit inline``, he gave the message that it returned zero rows but he tries to show the results like it have any to show, the columns name appears and all. I don't think that's correct. 

Another thing that idk if it's correct or not to appear is the button 'Show query box', the button appears on searchs that returns something and after my changes he doesn't appear in the first time that you do a search that gave zero results, but on the second one the buttons appears. That button is supposed to always appears or not?

I'm out right now, when I go home I'll be inserting some prints here.

**Comment by williamdes** [comment]
> That button is supposed to always appears or not?

For me, yes

**Comment by Webbithole** [comment]
Any updates on this?

**Comment by williamdes** [comment]
@liviuconcioiu would you help me with this one by a PR?

**Comment by liviuconcioiu** [comment]
> @liviuconcioiu would you help me with this one by a PR?

I'll give it a try!

**Comment by williamdes** [comment]
I checked `4.6.6` and it was the same behavior, and so was `3.5.3`. Same with `3.4.11.1`.
Due to the number of users that are annoyed by this, I merged  #19498 into 5.2.x
